### PR TITLE
rpc: Remove unused COINBASE_FLAGS

### PR DESCRIPTION
**Pull request by narula**
Commit d449772cf69c01932fc5d72c46054815d6300b3c stopped setting
COINBASE_FLAGS, and it looks like it hasn't been used since P2SH.

Following up on #17489, remove COINBASE_FLAGS which is unused. I verified that removing this did not change the contents of the coinbase's scriptSig.

### CODE REVIEW

File: src/rpc/mining.cpp [hunk 1]
```diff
@@ -607,7 +607,6 @@ static UniValue getblocktemplate(const JSONRPCRequest& request)
     }
 
     UniValue aux(UniValue::VOBJ);
-    aux.pushKV("flags", HexStr(COINBASE_FLAGS.begin(), COINBASE_FLAGS.end()));
```
Comment (MarcoFalke) [inline src/rpc/mining.cpp]: Not sure what the implications of removing this field are, as I don't have strong knowledge of the `getblocktemplate` BIP. Though, when this is removed, the documentation in `RPCHelpMan` should be updated as well. And maybe `aux` can be removed as well? 
cc @luke-jr

**Review by maflcko — APPROVED** [review]
ACK

File: src/rpc/mining.cpp (hunk 1 above)
Comment (luke-jr) [inline src/rpc/mining.cpp]: I'm not sure this counts as simple refactoring. Maybe rename the PR and make it a separate commit?

Comment (luke-jr) [reply src/rpc/mining.cpp]: It looks like "flags" is entirely missing in the BIPs.

Comment (luke-jr) [reply src/rpc/mining.cpp]: (Also not supported by either Eloipool nor libblkmaker)

Comment (MarcoFalke) [reply src/rpc/mining.cpp]: ok, but `coinbaseaux` is: https://en.bitcoin.it/wiki/BIP_0022#Specification

So I guess `aux` should stay but be kept empty.

Comment (MarcoFalke) [reply src/rpc/mining.cpp]: Done (renamed title)

**Comment by laanwj** [comment]
ACK 9aedabe67eedfee9c94c6a50962f11348eb99bca

**Comment by narula** [comment]
I can update RPCHelpMan. Is the right way to do that just remove `flags`, or would you prefer it to be marked in some way?

Does it make sense to remove the `coinbaseaux` flag as well?

**Comment by laanwj** [comment]
> I can update RPCHelpMan. Is the right way to do that just remove flags, or would you prefer it to be marked in some way?

To remove it, so that the help matches the output.

> Does it make sense to remove the coinbaseaux flag as well?

It's part of the BIP according to @luke-jr so I think it'd be better to keep it but leave it empty, as @MarcoFalke says.

**Comment by jnewbery** [comment]
utACK 9aedabe67eedfee9c94c6a50962f11348eb99bca once `flags` has been removed from the help text.

**Comment by narula** [comment]
Updated to remove `flags`.

**Comment by MarcoFalke** [comment]
ACK e9a27cf338dc618b8ecab8984abc54d588de8a05 💻

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK e9a27cf338dc618b8ecab8984abc54d588de8a05 💻
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUhjIQwAsMub4O7fZgenPS9fKJMwQxcxAS+hoZOFtDsGFN2pBFSCgB4oJE5c4u7T
OrTw53bux7AuRuhDjw7oTEp7Gv1J3IIR1dWP0Rnh5Pxjt2cPaoHElTHFWLqoR9tE
n2y7yTsqo2GSTze1JraibYW9CJY7v0ZS2SvrjCfhBR0ybYXKRKnGVFmPuVE2HwTl
jrBZklNRgL9ft/b0azemSXOuI8oy6OsBX56XOaiRGZ/RD7A32QspoePiSPqK1eqP
tHPEvH7BQkeGEVq2pTrdvMkkSbGplIZB3YI7TYzIyEGS2DWa9Kx/ERFCs1NFckP5
axmMc9DIgb+DYfPLx6Rj9YsY0CdQajRX9wmPGNIV04sUcwD19Q3gUE1Q14Ockq3u
9T4es3ppRDqvpXmMhuQMZy899NzLMrgtGx4qLJaFkSNTDZsA2JQdI3lD+uGuCu9D
x1a+ecpAg+UtSXTal3bIu861TgU0npszmJ+3OD37IknfA4vxeHTmPfo5xrYbPOSE
dc28Zx+w
=6gag
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `4af5f8707e419046118b148a954e705aa3ec85e3d60b38b9f3a0e1e32ec1cb31  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e8929401084af5f8707e419046118b148a954e705aa3ec85e3d60b38b9f3a0e1e32ec1cb31f01022c5eb8dd26682600d2a5b7fa2c5826008fff010308ea0cf00604ba8df2f19e3fb0212f708f1045dd6e11af0086c9017974cdaba3c0083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff0107ff73ca4248eb85c971045b032b7e5c208f1045dd6e11bf008fe0261ec3f5de6830083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6dfff010dc23337a5e3b11dd56eba23d145b432208f1045dd6e11af0085e469d8f2a3f30e20083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6df010f1562d0f5aef4c8291713aec19d696fd08f1045dd6e11af008a29041d4c64e6bb50083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>
